guardian/dotcom-rendering
# React Lazy for CMP

## What does this change?
Dynamically imports `ConsentManagementPlatform` from the ConsentManagementPlatform lib.

## Why?
`ConsentManagementPlatform` is not always used and we gains a major bundle reduction by dynamically loading it.

## How to test?
To avoid having to guess when CMP gets run, you can make sure the condition in `App.tsx` and `CMP.tsx` both return true and then you should be able to see the component being rendered. Look at the network tab in devtools to see that componet being loaded dynamically.

## Why React.lazy rather than `loadable-components`? I though we did SSR?
`loadable-components` can be used for SSR dynamic imports, but we don't gain any bundle reduction by doing this. We aim to reduce the frontend bundle (mainly `react.js`) so using React's native lazy API fits out need.

## doesn't `loadable-components` support code splitting libraries? 
Yes it does, but so does webpack, and unlike `loadable-components` we are not forced to wrap libraries in render-props.

## You are using Suspense, isn't that experimental?
Yes, this is a concern, the API could likely change. However the benefit right now is quite substantial, and the code changes needed to revert are quite minimal. 

## Link to supporting Trello card
https://trello.com/c/0qCpwbbO/1285-dynamic-imports
